R/build.r

```diff
@@ -20,15 +20,21 @@
 #'   line arguments to be passed to `R CMD build` if `binary = FALSE`,
 #'   or `R CMD install` if `binary = TRUE`.
 #' @param quiet if `TRUE` suppresses output from this function.
+#' @param needs_compilation Usually only needed if the packages has
+#'   C/C++/Fortran code. By default this is autodetected.
 #' @param compile_attributes if `TRUE` and the package uses Rcpp, call
-#'   [Rcpp::compileAttributes()] before building the package.
+#'   [Rcpp::compileAttributes()] before building the package. It is ignored
+#'   if package does not need compilation.
 #' @export
 #' @return a string giving the location (including file name) of the built
 #'  package
 build <- function(path = ".", dest_path = NULL, binary = FALSE, vignettes = TRUE,
-                  manual = FALSE, args = NULL, quiet = FALSE, compile_attributes = TRUE) {
+                  manual = FALSE, args = NULL, quiet = FALSE,
+                  needs_compilation = NA, compile_attributes = TRUE) {
```

Comment:
I wonder if the default here should be `NULL` rather than `NA`. I guess `NA` means the argument is always a `logical()`.

Reply:
YEah, usually `NULL` is the good default. This one might be different, though, because it is always a logical flag, and `NA` literally means that we don't know (yet).

Reply:
Yeah, that makes sense, I am good with this.